cmd/fortifyExecuteScan.go

```diff
@@ -0,0 +1,647 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/google/uuid"
+
+	"github.com/piper-validation/fortify-client-go/models"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/fortify"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/maven"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+
+	piperGithub "github.com/SAP/jenkins-library/pkg/github"
+)
+
+var auditStatus map[string]string
+
+const checkString = "<---CHECK FORTIFY---"
+const classpathFileName = "cp.txt"
+
+func fortifyExecuteScan(config fortifyExecuteScanOptions, telemetryData *telemetry.CustomData, influx *fortifyExecuteScanInflux) error {
+	auditStatus = map[string]string{}
+	sys := fortify.NewSystemInstance(config.ServerURL, config.APIEndpoint, config.AuthToken, time.Second*30)
+	c := command.Command{}
+	// reroute command output to loging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+	c.SetEnv(os.Environ())
+	return runFortifyScan(config, sys, &c, telemetryData, influx)
+}
+
+func runFortifyScan(config fortifyExecuteScanOptions, sys fortify.System, command execRunner, telemetryData *telemetry.CustomData, influx *fortifyExecuteScanInflux) error {
+	log.Entry().Debugf("Running Fortify scan against SSC at %v", config.ServerURL)
+	var gav piperutils.BuildDescriptor
+	var err error
+	if config.ScanType == "maven" {
+		gav, err = piperutils.GetMavenCoordinates(config.BuildDescriptorFile)
+	}
+	if config.ScanType == "pip" {
+		gav, err = piperutils.GetPipCoordinates(config.BuildDescriptorFile)
+	}
+	if err != nil {
+		log.Entry().Warnf("Unable to load project coordinates from descriptor %v: %w", config.BuildDescriptorFile, err)
+	}
+	fortifyProjectName, fortifyProjectVersion := piperutils.DetermineProjectCoordinates(config.ProjectName, config.ProjectVersion, config.DefaultVersioningModel, gav)
+	project, err := sys.GetProjectByName(fortifyProjectName, config.AutoCreate, fortifyProjectVersion)
+	if err != nil {
+		log.Entry().Fatalf("Failed to load project %v: %w", fortifyProjectName, err)
```

Comment:
Exiting via `Fatal()` here makes it more cumbersome to test, IMHO. The function is supposed to return an error which is currently returned from `fortifyExecuteScan`, but that is nowhere tested. The new pattern is to return nothing from `fortifyExecuteScan`, and instead check the error from `runFortifyExecuteScan` and then exit via the logger. Same comment of course to the other places with `Fatal()` here.